lib/elixir/lib/keyword.ex

```diff
@@ -186,6 +186,193 @@ defmodule Keyword do
     :lists.foldl(fun, [], Enum.reverse(pairs))
   end
 
+  @doc """
+  Ensures the first argument is a `keyword` with the given
+  keys and default values.
+  The second argument must be a list of atoms, specifying
+  a given key, or tuples specifying a key and a default value.
+  If any of the keys in the `keyword` is not defined on
+  `values`, it raises an error.
```

Comment:
This doc is almost the same as the one for `validate/2`. Is there a best-practices approach to avoid being this repetitive, or is the way this was done fine?

Reply:
perhaps something like this?

```
  @doc """
  Similar to `validate/2` but returns the keyword or raises an error.

  ## Examples

      iex> Keyword.validate!([two: 3], [one: 1, two: 2])
      [one: 1, two: 3]

      iex> Keyword.validate!([three: 3], [one: 1, two: 2])
      ** (ArgumentError) unknown key :three in [three: 3], expected one of [:one, :two]

  """
```

btw,  I'd define the `validate!/2` function _after_ `validate/2` given they'd show up in docs in that order